client/asset/eth/eth.go

```diff
@@ -677,13 +688,32 @@ func (eth *ExchangeWallet) Redeem(form *asset.RedeemForm) ([]dex.Bytes, asset.Co
 		return fail(errors.New("Redeem: must be called with at least 1 redemption"))
 	}
 
+	var contractVersion uint32 // require a consistent version since this is a single transaction
 	inputs := make([]dex.Bytes, 0, len(form.Redemptions))
 	var redeemedValue uint64
-	for _, redemption := range form.Redemptions {
-		var secretHash, secret [32]byte
-		copy(secretHash[:], redemption.Spends.SecretHash)
+	for i, redemption := range form.Redemptions {
+		// NOTE: redemption.Spends.SecretHash is a dup of the hash extracted
+		// from redemption.Spends.Contract. Even for scriptable UTXO assets, the
+		// redeem script in this Contract field is redundant with the SecretHash
+		// field as ExtractSwapDetails can be applied to extract the hash.
+		ver, secretHash, err := dexeth.DecodeContractData(redemption.Spends.Contract)
```

Comment:
Note that `AuditContract` is still in PR, but see also my comment: https://github.com/decred/dcrdex/pull/1319/files#r764101981